src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2362,37 +2362,81 @@ public override BoundNode VisitConditionalAccess(BoundConditionalAccess node)
 
         public override BoundNode VisitConditionalOperator(BoundConditionalOperator node)
         {
-            var isByRef = node.IsRef;
+            bool isConstantTrue = IsConstantTrue(node.Condition);
+            bool isConstantFalse = IsConstantFalse(node.Condition);
 
             VisitCondition(node.Condition);
             var consequenceState = this.StateWhenTrue;
             var alternativeState = this.StateWhenFalse;
 
+            TypeWithState consequenceRValue;
+            TypeWithState alternativeRValue;
+
+            if (node.IsRef)
+            {
+                TypeWithAnnotations consequenceLValue;
+                TypeWithAnnotations alternativeLValue;
+
+                if (isConstantTrue)
```

Comment:
>if [](start = 16, length = 2)

This whole series of if-then-else should be replaceable by the following (because `VisitLvalueWithAnnotations` unsplits):
``` c#
                    (consequenceLValue, consequenceRValue) = visitConditionalRefOperand(consequenceState, node.Consequence);
                    consequenceState = this.State;
                    (alternativeLValue, alternativeRValue) = visitConditionalRefOperand(alternativeState, node.Alternative);
                    Join(ref this.State, ref consequenceState);
```